### PR TITLE
Created database tables, associations, and rake sample_data task

### PR DESCRIPTION
**Pull request by jt-is-coding**
Maybe should have done part 1 on a separate branch, oops!

### CODE REVIEW

**Review by jb520 — APPROVED** [review]
Looking good Jeremy!

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Foreign Keys
+#
+#  fk_rails_...  (fan_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Like < ApplicationRecord
+  belongs_to :fan, class_name: "User", counter_cache: true
+  belongs_to :photo, counter_cache: true
+
+  # validates :user_id, uniqueness: { scope: :photo_id, message: "has already liked this photo" }
```
Comment (thierrychau) [inline app/models/like.rb:25]: We do want this validation. If you were getting an error, let's walk through this. Validation is a constraint on a field, here you would have a constraint on a field called `user_id`. Is there a field `user_id` in this model?

**Review by thierrychau — CHANGES_REQUESTED** [review]
Hi Jeremy, great job so far. I have minor comments, can you look at this and fix it? Thanks!

File: app/models/like.rb (hunk 1 above)
Comment (jt-is-coding) [reply app/models/like.rb]: Ah oops! I forgot I commented that out to verify that's where the issue was stemming from. I went ahead and swapped user_id to fan_id, as well as putting some text strings at the end of the sample_data task to verify that everything is working as expected.

**Review by thierrychau — APPROVED** [review]
Great job, good to merge!

**Pull request merged**